xlsxwriter/worksheet.py

```diff
@@ -418,6 +419,11 @@ def write(self, row, col, *args):
             return self.write_number(row, col, *args)
         if token_type is str:
             return self.write_token_as_string(token, row, col, *args)
+        if token_type in (datetime.datetime,
```

Comment:
It would be cleaner to change the `supported_datetime()` method from using `isinstance()` to `type()` and it wouldn't have to be called further down. Would an additional function call cause you much of a performance penalty?